### PR TITLE
Introduce Testcontainers & Flyway

### PR DESCRIPTION
**Pull request by otrosien**
Adds Testcontainers to the Fahrschein-JDBC and Fahrschein-Redis modules. I'm not adding a re-usable docker-compose based setup yet, because the Integration Tests are so far about testing writing cursors into an external storage like PostgreSQL or Redis, and not yet about connecting to Nakadi. This will come in another PR.

## Changes:
* Removes disambiguation between integration and unit tests (can be re-introduced once needed)
* Moves Persistence provisioning and initialization to Testcontainers / Flyway
* Makes SQLs into flyway-compatible directory setup
* Adapts README documentation
 
Relates to https://github.com/zalando-nakadi/fahrschein/issues/271

### CODE REVIEW

**Comment by othon2000** [comment]
@otrosien about removing the DDL scripts from `src/main/resources:

If an application uses `JdbcCursorManager`, it needs to create the corresponding DB schema (and functions). 

I searched [all of Zalando](https://github.bus.zalan.do/search?q=JdbcCursorManager&type=code) and [all of github](https://github.com/search?q=JdbcCursorManager&type=code) but could not find any usage of it. [1]

However, if I were to find an usage, **I would expect the application to load the DDL scripts from the classpath** even though this is not documented anywhere.

That said, if we proceed with this PR, I recommend to move the scripts back and to document the usage of Flyway as part of using `fahrschein-jdbc`.

However, because of what I have mentioned in [1], does it even make sense to maintain this module? Why not removing it instead? Same goes for `RedisCursorManager`.

**Comment by otrosien** [comment]
I'm afraid of breaking people's applications because of "too-much-magic". Suddenly Flyway finds new migrations on the classpath, and tries to execute them... Not really something I would expect when updating a dependency.

**Comment by otrosien** [comment]
It's also nothing I found anywhere else. Flyway migrations should be in control of the application, which means we can point to them in the README, but people need to actively copy them over.

**Comment by othon2000** [comment]
> It's also nothing I found anywhere else. Flyway migrations should be in control of the application, which means we can point to them in the README, but people need to actively copy them over.

Sorry I was not very clear. I did not propose to embed flyway and do the migrations without the control of the user, but instead to **document how the user should do it**. Thus the migration scripts would need to be in the classpath (as they are today).

**Comment by othon2000** [comment]
:+1:

**Review by othon2000 — CHANGES_REQUESTED** [review]
The PR description also needs to be updated, regarding:

> Why have we been delivering the SQLs in src/main/resources, is that something genuinely relevant for our clients? I assume not, so I moved them into src/test/resources.

File: fahrschein-jdbc/src/test/java/org/zalando/fahrschein/jdbc/LocalPostgresConfiguration.java [hunk 1]
```diff
@@ -16,13 +18,23 @@
 @Configuration
 @EnableTransactionManagement(proxyTargetClass = true)
 public class LocalPostgresConfiguration {
+
+    public static PostgreSQLContainer db = new PostgreSQLContainer("postgres:13")
+            .withDatabaseName("local_nakadi_cursor_db");
+
+    static {
+        db.start();
```
[the remark this answers is not in the document]
Comment (otrosien) [reply fahrschein-jdbc/src/test/java/org/zalando/fahrschein/jdbc/LocalPostgresConfiguration.java:26]: I will add one for the nakadi-docker PR. A simple postgres (or a redis) is very easy and fast to spin up, which saves the hassle of having to think about dirty state.

**Comment by othon2000** [comment]
:+1:

**Comment by otrosien** [comment]
👍

**Pull request merged**